src/app/outbound/endpoint.rs

```diff
@@ -19,6 +21,9 @@ pub struct Endpoint {
     pub http_settings: settings::Settings,
 }
 
+#[derive(Copy, Clone, Debug)]
+pub struct FromMetadata;
```

Comment:
Nit/TIOLI: If there was an implementation of `MapEndpoint<A, B>` for `F where F: Fn(&A, SocketAddr, B) -> Out`, I think this could just be a function, and we could reference it as a function pointer instead of an empty struct?

Reply:
i like having the thing be nameable. is there a problem as it is?

Reply:
nope, that was a TIOLI, and i agree that having a name for it is valuable. carry on!